File: domain/src/main/java/by/rekuts/travelagency/domain/aspects/LoggingAspect.java
package by.rekuts.travelagency.domain.aspects;

import lombok.extern.slf4j.Slf4j;
import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.annotation.After;
import org.aspectj.lang.annotation.AfterReturning;
import org.aspectj.lang.annotation.Aspect;
import org.springframework.stereotype.Component;

@Slf4j
@Aspect
@Component
public class LoggingAspect {

    @After("execution(* *.insert(..))")
    public void log(JoinPoint point) {
        log.info("!!!!!!!!!ASPECT-J-INSERT!!!!!!!!!!!" + point.getSignature().getName() + " called...");
    }

    @AfterReturning(value = "@annotation(by.rekuts.travelagency.domain.aspects.LogReturn))", returning = "returnValue")
    public void logResult(JoinPoint joinPoint, Object returnValue) {
        log.info("!!!!!!!!!!!!!!!!ASPECT-J-RETURN!!!!!!!!!!!!!!!!" + joinPoint.getSignature().getName() +
                " called... " + returnValue);
    }
}

File: domain/src/main/scala/by/rekuts/travelagency/domain/aspects/LoggingAspect.scala
package by.rekuts.travelagency.domain.aspects

import org.aspectj.lang.JoinPoint
import org.aspectj.lang.annotation.{After, AfterReturning, Aspect}
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.stereotype.Component

@Aspect
@Component
class LoggingAspect {
  @After("execution(* *.insert(..))")
  def log(point: JoinPoint): Unit = {
    LoggingAspect.log.info("!!!!!!!!!ASPECT-J-INSERT!!!!!!!!!!!" + point.getSignature.getName + " called...")
  }

  @AfterReturning(value = "@annotation(by.rekuts.travelagency.domain.aspects.LogReturn))", returning = "returnValue")
  def logResult(joinPoint: JoinPoint, returnValue: Any): Unit = {
    LoggingAspect.log.info("!!!!!!!!!!!!!!!!ASPECT-J-RETURN!!!!!!!!!!!!!!!!" + joinPoint.getSignature.getName + " called... " + returnValue)
  }

  object LoggingAspect {
    val log: Logger = LoggerFactory.getLogger(classOf[LoggingAspect])
  }
}
